### PR TITLE
build: update spdkrs, dependencies, nixpkgs, cargo

### PR DESCRIPTION
**Pull request by tiagolobocastro**
A lot of code changes, related to clippy more strict rules.

### CODE REVIEW

**Comment by tiagolobocastro** [comment]
bors try

**Comment by tiagolobocastro** [comment]
bors try

**Comment by tiagolobocastro** [comment]
Eh, looks like fs tools are too new for the CI systems:
```
[17548489.305395] XFS (nvme2n1): Superblock has unknown incompatible features (0x20) enabled.
[17548489.305396] XFS (nvme2n1): Filesystem cannot be safely mounted by this kernel.
[17548489.305399] XFS (nvme2n1): SB validate failed with error -22.
```
Looks like missing the nrext64 feature, added on 5.13 kernel

**Comment by tiagolobocastro** [comment]
bors try

**Comment by tiagolobocastro** [comment]
bors try

**Comment by tiagolobocastro** [comment]
bors merge

File: io-engine/src/bdev/nvmx/controller.rs [hunk 1]
```diff
@@ -911,6 +911,7 @@ pub(crate) async fn destroy_device(name: String) -> Result<(), BdevError> {
             // Release the lock before waiting for controller shutdown.
             drop(controller);
 
+            // clippy bug...
```
Comment (Abhinandan-Purkait) [inline io-engine/src/bdev/nvmx/controller.rs]: Is this comment needed?

File: io-engine/src/bin/io-engine-client/main.rs [hunk 2]
```diff
@@ -22,37 +22,46 @@ pub enum ClientError {
     #[snafu(display("gRPC status: {}", source))]
     GrpcStatus {
         source: tonic::Status,
-        backtrace: Backtrace,
+        backtrace: Option<Backtrace>,
```
Comment (Abhinandan-Purkait) [inline io-engine/src/bin/io-engine-client/main.rs:25]: Do we need to make these options?

File: io-engine/src/bin/io-engine-client/main.rs [hunk 3]
```diff
@@ -22,37 +22,46 @@ pub enum ClientError {
...
+    {
         "v0" => v0::main_().await,
         "v1" => v1::main_().await,
         "" => v1::main_().await,
         version => {
             panic!("Invalid Api version set: {}", version)
         }
+    };
+    if let Err(error) = result {
+        eprintln!("{}", error);
```
Comment (Abhinandan-Purkait) [inline io-engine/src/bin/io-engine-client/main.rs:60]: Why not the tracing error?

File: io-engine/src/core/fault_injection/mod.rs [hunk 4]
```diff
@@ -34,7 +34,7 @@ pub enum FaultDomain {
     NexusChild,
     /// Fault injection on block device abstraction level.
     BlockDevice,
-    ///
+    /// TODO
```
Comment (Abhinandan-Purkait) [inline io-engine/src/core/fault_injection/mod.rs:37]: Did you plan to do this and missed by chance ?

File: io-engine/src/grpc/v0/nexus_grpc.rs [hunk 5]
```diff
@@ -1,4 +1,4 @@
-///! Helpers related to nexus grpc methods.
+//! Helpers related to nexus grpc methods.
```
Comment (Abhinandan-Purkait) [inline io-engine/src/grpc/v0/nexus_grpc.rs:1]: Ohh but ///! is for doc comment right or was it //!?

File: io-engine/src/grpc/server.rs [hunk 6]
```diff
@@ -67,7 +67,7 @@ impl MayastorGrpcServer {
         rpc_addr: String,
         api_versions: Vec<ApiVersion>,
     ) -> Result<(), ()> {
-        let mut rcv_chan = Self::get_or_init().rcv_chan.clone();
+        let mut rcv_chan = Box::pin(Self::get_or_init().rcv_chan.clone());
```
Comment (Abhinandan-Purkait) [inline io-engine/src/grpc/server.rs:70]: Ohh do we want to use heap now?

File: io-engine/src/lvs/lvol_snapshot.rs [hunk 7]
```diff
@@ -106,6 +106,8 @@ pub trait LvolSnapshotOps {
 
     /// create replica snapshot inner function to call spdk snapshot create
     /// function.
+    /// # Safety
+    /// TODO
```
Comment (Abhinandan-Purkait) [inline io-engine/src/lvs/lvol_snapshot.rs:110]: Did you intent to make any changes here?

File: io-engine/src/lvs/lvs_lvol.rs [hunk 8]
```diff
@@ -615,6 +616,8 @@ pub trait LvsLvol: LogicalVolume + Share {
     ) -> Option<*mut spdk_blob>;
 
     /// Get the next spdk_blob from the parent blob.
+    /// # Safety
+    /// TODO
```
Comment (Abhinandan-Purkait) [inline io-engine/src/lvs/lvs_lvol.rs:620]: Did you intent to make any changes here?

File: test/python/common/nvme.py [hunk 9]
```diff
@@ -68,6 +68,7 @@ def nvme_connect(uri, delay=10, tmo=600):
     command = (
         f"nix-sudo nvme connect -t tcp -s {port} -a {host} -n {nqn} -c {delay} -l {tmo}"
     )
+    print(command)
```
Comment (Abhinandan-Purkait) [inline test/python/common/nvme.py:71]: I see a lot of print, was it for debug or is it intentional.

**Review by Abhinandan-Purkait — APPROVED** [review]
Thanks. Just pointed out the todos in case that was meant to be added.

**Review by niladrih — APPROVED** [review]

File: io-engine/src/bdev/nvmx/controller.rs (hunk 1 above)
Comment (tiagolobocastro) [reply io-engine/src/bdev/nvmx/controller.rs]: Oh no, I was using wrong clippy I think, then reverted the change but left the comment

File: io-engine/src/bin/io-engine-client/main.rs (hunk 2 above)
Comment (tiagolobocastro) [reply io-engine/src/bin/io-engine-client/main.rs:25]: Yes, semantics change from the snafu

File: io-engine/src/bin/io-engine-client/main.rs (hunk 3 above)
Comment (tiagolobocastro) [reply io-engine/src/bin/io-engine-client/main.rs:60]: to keep the same format

File: io-engine/src/core/fault_injection/mod.rs (hunk 4 above)
Comment (tiagolobocastro) [reply io-engine/src/core/fault_injection/mod.rs:37]: Now it's required to add comment text, but I don't know what to write for it, so left it like so

File: io-engine/src/grpc/server.rs (hunk 6 above)
Comment (tiagolobocastro) [reply io-engine/src/grpc/server.rs:70]: Api changed from the crate, now we have to do it ourselves

File: io-engine/src/lvs/lvol_snapshot.rs (hunk 7 above)
Comment (tiagolobocastro) [reply io-engine/src/lvs/lvol_snapshot.rs:110]: Yes, it's required by clippy on unsafe fn's now

File: test/python/common/nvme.py (hunk 9 above)
Comment (tiagolobocastro) [reply test/python/common/nvme.py:71]: It's for debug but let's keep it, it doesn't get output unless the test fails anyway

File: io-engine/src/grpc/v0/nexus_grpc.rs (hunk 5 above)
Comment (tiagolobocastro) [reply io-engine/src/grpc/v0/nexus_grpc.rs:1]: It's `//!` it seems

**Comment by tiagolobocastro** [comment]
bors merge

**Pull request merged**